core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2690,6 +2690,48 @@ private static class ContainsSubstrImplementor extends AbstractRexCallImplemento
     }
   }
 
+  /** Implementor for MYSQL {@code BIT_COUNT} function. */
+  private static class BitCountMySQLImplementor extends AbstractRexCallImplementor {
+    BitCountMySQLImplementor() {
+      super("bitCount", NullPolicy.STRICT, false);
+    }
+
+    @Override Expression implementSafe(final RexToLixTranslator translator,
+        final RexCall call, final List<Expression> argValueList) {
+      Expression expr = argValueList.get(0);
+      RelDataType relDataType = call.getOperands().get(0).getType();
+      if (SqlTypeUtil.isNull(relDataType)) {
+        return argValueList.get(0);
+      }
+      // In MySQL, BIT_COUNT(TIMESTAMP '1996-08-03 16:22:34') is converted to
+      // BIT_COUNT('19960803162234') for calculation, so the internal int value
+      // needs to be converted to DATE/TIME and TIMESTAMP.
+      SqlTypeName type = relDataType.getSqlTypeName();
+      switch (type) {
+      case VARBINARY:
+      case BINARY:
+        return Expressions.call(SqlFunctions.class, "bitCount", expr);
+      case DATE:
+        expr = Expressions.call(BuiltInMethod.INTERNAL_TO_DATE.method, expr);
+        break;
+      case TIME:
+      case TIME_WITH_LOCAL_TIME_ZONE:
+      case TIME_TZ:
+        expr = Expressions.call(BuiltInMethod.INTERNAL_TO_TIME.method, expr);
+        break;
+      case TIMESTAMP:
+      case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
```

Comment:
Perhaps tests with the local timezone are difficult, but I don't see tests with timezone either.
It would be great to cover all these cases with tests.

Reply:
Hi @mihaibudiu, I checked the documentation of MySQL TIMESTAMP - https://dev.mysql.com/doc/refman/8.4/en/datetime.html, and found that it does not provide syntax for `with time zone` and `with local time zone`. Does this mean I should remove `TIMESTAMP_WITH_LOCAL_TIME_ZONE` and `TIMESTAMP_TZ` here?

Reply:
It's certainly simpler not to support them